### PR TITLE
create subject

### PR DESCRIPTION
**Pull request by VuongLong**

### CODE REVIEW

**Comment by binhthanhnguyen** [comment]
ok

**Comment by thiennv57** [comment]
OK

File: app/controllers/admin/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Admin::SubjectsController < ApplicationController
+  def create
+    @subject = Subject.new subject_params
+    if @subject.save
+      flash.now[:success] = t("admin.create.success")
```
Comment (ThuBM) [inline app/controllers/admin/subjects_controller.rb]: remove ()

File: app/controllers/admin/subjects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      redirect_to admin_subjects_path 
+    else
+      flash.now[:failed] = t("admin.create.fail")
```
Comment (ThuBM) [inline app/controllers/admin/subjects_controller.rb]: remove ()

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -3,4 +3,24 @@ def full_title page_title = ""
...
+    end
+    link_to_function(name, "add_fields(this, \"#{association}\", \"#{escape_javascript(fields)}\")")
+  end
+
+  def link_to_function name, function, html_options={}
+    message = t "message-link-to-function"
+    ActiveSupport::Deprecation.warn message
+    onclick = "#{"#{html_options[:onclick]}; " if html_options[:onclick]}#{function}; return false;"
+    href = html_options[:href] || '#'
+    content_tag(:a, name, html_options.merge(href: href, onclick: onclick))
```
Comment (ThuBM) [inline app/helpers/application_helper.rb]: remove first ()

File: app/assets/stylesheets/custom.css.scss [hunk 4]
```diff
@@ -74,3 +74,11 @@ footer {
     }
   }
 }
+
+#nav_subject {
+  line-height:30px;
+  background-color:#eeeeee;
+  width:300px;
+  float:left;
+  padding:5px;
```
Comment (ThuBM) [inline app/assets/stylesheets/custom.css.scss]: add space after :   in above 5 lines

**Comment by ThuBM** [comment]
OK

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -3,4 +3,24 @@ def full_title page_title = ""
...
+
+  def link_to_add_fields name, f, association
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object, child_index: "new_#{association}") do |builder|
+      render(association.to_s.singularize + "_fields", f: builder)
+    end
+    link_to_function(name, "add_fields(this, \"#{association}\", \"#{escape_javascript(fields)}\")")
+  end
+
+  def link_to_function name, function, html_options={}
```
Comment (dieunb) [inline app/helpers/application_helper.rb]: ` html_options = {}`

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
   root "static_pages#home"
+
+  namespace :admin do
```
Comment (dieunb) [inline config/routes.rb]: admin root path

Comment (VuongLong) [inline config/routes.rb]: i've fixed it below that line

**Pull request merged**